### PR TITLE
Implement ETag support for server rendered pages.

### PR DESCRIPTION
**Pull request by arunoda**
Fixes https://github.com/zeit/next.js/issues/716

Here we use strong etags. That's because, there's no big perf. difference between strong and weak etag generation.
See: https://github.com/jshttp/etag#benchmark

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,8 +61,10 @@
     "case-sensitive-paths-webpack-plugin": "2.0.0",
     "cross-spawn": "5.1.0",
     "del": "2.2.2",
+    "etag": "1.8.0",
+    "fresh": "0.5.0",
     "friendly-errors-webpack-plugin": "1.5.0",
-    "glob": "^7.1.1",
+    "glob": "7.1.1",
```
Comment (arunoda) [inline package.json]: This is not related to this PR. But just updated.

**Comment by impronunciable** [comment]
Looks good to me @arunoda
